### PR TITLE
8259799: vmTestbase/nsk/jvmti/Breakpoint/breakpoint001 is incorrect

### PR DESCRIPTION
**Pull request by lmesnik**
est vmTestbase/nsk/jvmti/Breakpoint/breakpoint001 has incorrect check of strcmp results here:

  for (i=0; i<METH_NUM; i++)
        if (strcmp(methNam,METHODS[i][0]) &&
                strcmp(methSig,METHODS[i][1])) {
            printf("CHECK PASSED: method name: \"%s\"\tsignature: \"%s\" %d\n",
                   methNam, methSig, i);
            if (checkStatus == PASSED)
                bpEvents[i]++;
            break;
        }

So test passed when both strcmp (name,sig) are not zero.

The test passes only because there are 2 methods that are checked and it increases counters for "incorrect" methods.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259799](https://bugs.openjdk.java.net/browse/JDK-8259799): vmTestbase/nsk/jvmti/Breakpoint/breakpoint001 is incorrect

### Reviewers
 * [Igor Ignatyev](https://openjdk.java.net/census#iignatyev) (@iignatev - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2084/head:pull/2084`
`$ git checkout pull/2084`

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2003, 2019, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2003, 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (iignatev) [inline test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp:2]: it's 2021

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp [hunk 2]
```diff
@@ -170,8 +170,8 @@ Breakpoint(jvmtiEnv *jvmti_env, JNIEnv* jni_env, jthread thread,
     }
 
     for (i=0; i<METH_NUM; i++)
-        if (strcmp(methNam,METHODS[i][0]) &&
-                strcmp(methSig,METHODS[i][1])) {
+        if (strcmp(methNam,METHODS[i][0]) == 0 &&
```
Comment (iignatev) [inline test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp:173]: could you please add space before `,`?

**Review by iignatev — APPROVED** [review]
@lmesnik , looks good to me, besides a few nits.

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp (hunk 2 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp]: done

File: test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp (hunk 1 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/jvmti/Breakpoint/breakpoint001/breakpoint001.cpp]: ough, thanks!

**Review by iignatev — APPROVED** [review]

**Review by sspitsyn — APPROVED** [review]
Hi Leonid,
LGTM++
Thanks,
Serguei

**Review by plummercj — APPROVED** [review]

**Pull request closed without merging**